consensus/tool/src/main.rs

```diff
@@ -0,0 +1,156 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! A cli tool for interrogating a mobilecoin consensus node.
+
+#![deny(missing_docs)]
+
+use clap::{Parser, Subcommand};
+use grpcio::{ChannelBuilder, EnvBuilder};
+use mc_common::logger::{create_app_logger, o};
+use mc_consensus_api::{consensus_common_grpc::BlockchainApiClient, empty::Empty};
+use mc_util_grpc::ConnectionUriGrpcioChannel;
+use mc_util_uri::ConsensusClientUri;
+use serde_json::{json, to_string_pretty, Map, Value};
+use std::{sync::Arc, time::Duration};
+
+/// Command line config, set with defaults that will work with
+/// a standard mobilecoind instance
+#[derive(Clone, Debug, Parser)]
+#[clap(
+    name = "mc-consensus-tool",
+    about = "A cli tool for interrogating one or more consensus nodes"
+)]
+pub struct Config {
+    /// Command to use
+    #[clap(subcommand)]
+    pub tool_command: ToolCommand,
+
+    /// Consensus_uri's to connect to
+    #[clap(global = true)]
+    pub consensus_uris: Vec<ConsensusClientUri>,
+}
+
+/// Commands that the tool recognizes
+#[derive(Clone, Debug, Subcommand)]
+pub enum ToolCommand {
+    /// Status: Prints the network status, including block count and block
+    /// version
+    Status,
+    /// Wait-for-block: Blocks until all specified nodes have externalized the
+    /// given block index.
+    WaitForBlock {
+        /// Block index which must appear
+        index: u64,
+        /// Polling period in seconds
+        #[clap(long, default_value = "1")]
+        period: u64,
```

Comment:
Optional/Equivalent: Make this a `Duration`.

I'd been meaning to replace (the body of) `mc_util_parse::parse_duration_in_seconds` with [`parse_duration`](https://docs.rs/parse_duration/), which would make for a nicer CLI IMO, by letting users specify durations like `--period=10m`
(You certainly don't have to do this; this is just a CBB)